packages/zoe/src/contractFacet/zcfMint.js

```diff
@@ -25,45 +24,59 @@ export const makeZCFMintFactory = (
   /** @type {ZcfMintReallocator} */ reallocator,
 ) => {
-  /** @type {SetStore<import('@agoric/vat-data').Baggage>} The set of baggages for zcfMints */
-  const zcfMintBaggageSet = provideDurableSetStore(zcfBaggage, 'baggageSet');
-
   /**
-   * retrieve the state of the zcfMint from the baggage, and create a durable
-   * singleton reflecting that state.
-   *
    * @param {import('@agoric/vat-data').Baggage} zcfMintBaggage
    */
-  const provideDurableZcfMint = zcfMintBaggage => {
-    const keyword = zcfMintBaggage.get('keyword');
-    const zoeMint = zcfMintBaggage.get('zoeMint');
-    const {
-      brand: mintyBrand,
-      issuer: mintyIssuer,
-      displayInfo: mintyDisplayInfo,
-    } = zcfMintBaggage.get('issuerRecord');
-    const mintyIssuerRecord = makeIssuerRecord(
-      mintyBrand,
-      mintyIssuer,
-      mintyDisplayInfo,
-    );
-    recordIssuer(keyword, mintyIssuerRecord);
-
-    const empty = AmountMath.makeEmpty(mintyBrand, mintyDisplayInfo.assetKind);
-    const add = (total, amountToAdd) =>
-      AmountMath.add(total, amountToAdd, mintyBrand);
-
-    return prepareSingleton(
+  const prepareDurableZcfMint = zcfMintBaggage => {
+    return prepareExoClass(
       zcfMintBaggage,
       'zcfMint',
-      /** @type {ZCFMint} */
+      M.interface('ZCFMint', {
+        getIssuerRecord: M.call().returns(IssuerRecordShape),
+        mintGains: M.call(AmountKeywordRecordShape, M.any()).returns(SeatShape),
+        burnLosses: M.call(AmountKeywordRecordShape, M.any()).returns(
+          SeatShape,
+        ),
+      }),
+      (
+        /** @type {string} */ keyword,
+        /** @type {ZoeMint} */ zoeMint,
+        /** @type {IssuerRecord} */ issuerRecord,
+      ) => {
+        const {
+          brand: mintyBrand,
+          issuer: mintyIssuer,
+          displayInfo: mintyDisplayInfo,
+        } = issuerRecord;
+        if (!mintyDisplayInfo) {
+          throw Fail`DurableZcfMint requires display info`;
+        }
+        const mintyIssuerRecord = makeIssuerRecord(
+          mintyBrand,
+          mintyIssuer,
+          mintyDisplayInfo,
+        );
+        recordIssuer(keyword, mintyIssuerRecord);
+
+        const empty = AmountMath.makeEmpty(
+          mintyBrand,
+          mintyDisplayInfo.assetKind,
+        );
+
+        return { empty, keyword, zoeMint, mintyIssuerRecord };
+      },
       {
         getIssuerRecord: () => {
-          return mintyIssuerRecord;
+          return this.state.mintyIssuerRecord;
         },
         /** @type {(gains: Record<string, Amount>, zcfSeat?: ZCFSeat) => ZCFSeat} */
         mintGains: (gains, zcfSeat = makeEmptySeatKit().zcfSeat) => {
+          const { empty, mintyBrand, zoeMint } = this.state;
           gains = coerceAmountKeywordRecord(gains, getAssetKindByBrand);
+          const add = (
+            /** @type {Amount<AssetKind>} */ total,
+            /** @type {Amount<AssetKind>} */ amountToAdd,
```

Comment:
I don't see how these types help when `add`'s only use is to be passed to `.reduce()`.  `gains` is already typed because it is externally visible.

Are they here out of habit, or do they improve maintainability somehow?

Reply:
interim state. a subsequent commit had,
```js
        /** @type {(total: Amount, amountToAdd: Amount) => Amount}  */
        const add = (total, amountToAdd) =>
          AmountMath.add(total, amountToAdd, mintyBrand);
```

the type declaration doesn't add much but imo it's still better